src/main/java/com/softserve/academy/spaced/repetition/controller/DeckRatingController.java

```diff
@@ -0,0 +1,46 @@
+package com.softserve.academy.spaced.repetition.controller;
+import com.softserve.academy.spaced.repetition.DTO.DTO;
+import com.softserve.academy.spaced.repetition.DTO.DTOBuilder;
+import com.softserve.academy.spaced.repetition.DTO.impl.DeckRatingPublicDTO;
+import com.softserve.academy.spaced.repetition.domain.DeckRating;
+import com.softserve.academy.spaced.repetition.exceptions.NotAuthorisedUserException;
+import com.softserve.academy.spaced.repetition.exceptions.RatingsBadValueException;
+import com.softserve.academy.spaced.repetition.service.DeckRatingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Link;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+public class DeckRatingController {
+
+    public static final int MIN_RATING = 1;
+    public static final int MAX_RATING = 5;
+
+    @Autowired
+    private DeckRatingService deckRatingService;
+
+    @GetMapping("api/rate/deck/{id}")
+    public ResponseEntity<DeckRatingPublicDTO> getDeckRatingById(@PathVariable Long deckId) {
+        DeckRating deckRating = deckRatingService.getDeckRatingById(deckId);
+        Link selfLink = linkTo(methodOn(com.softserve.academy.spaced.repetition.controller.DeckRatingController.class).getDeckRatingById(deckRating.getId())).withRel("deckRating");
+        DeckRatingPublicDTO deckRatingDTO = DTOBuilder.buildDtoForEntity(deckRating, DeckRatingPublicDTO.class, selfLink);
+        return new ResponseEntity<>(deckRatingDTO, HttpStatus.OK);
+    }
+
+    @PostMapping("/api/private/deck/{deckId}")
+    public ResponseEntity<DTO<DeckRating>> addDeckRating(@RequestBody DeckRating deckRating, @PathVariable Long courseId) throws RatingsBadValueException, NotAuthorisedUserException {
```

Comment:
Why does the controller accept the entity and not the dto?

Reply:
What is dto, and why we have to use dto here???

Reply:
Lets leave it as is, because as I see all the controllers follow this pattern.

In multilayer architecture the controller should not be aware of the repositories and enties. Imagine one day you decide to change the persistence framework, to replace Hibernate by some noSQL storage. You replace all repositories by something else. In that case all services have to be changed. If controllers are aware of entities they too have be changed.